sapi/fpm/fpm/fpm_main.c

```diff
@@ -1908,6 +1908,8 @@ consult the installation file that came with this distribution, or visit \n\
 				 * so cleanup and continue, request shutdown is
 				 * handled later */
 
+				zend_destroy_file_handle(&file_handle);
```

Comment:
I don't think we need this one as this is after failure which should release everything before? Or do you see some leak here?

Reply:
I'm not sure I understand what you mean. Won't the while loop keep accepting new requests?

Reply:
This is executed if `php_fopen_primary_script(&file_handle) == FAILURE` which means that `file_handle` should not get initialized here and there should be no need to free it. I think it should be freed only for SUCCESS as you do below unless I missed something?

Reply:
Oops. Yes, you're correct. Idk what I was thinking here...